packages/core/lib/command-utils.js

```diff
@@ -201,6 +203,37 @@ const runCommand = async function (command, options) {
   return await command.run(options);
 };
 
+const processHelpInput = inputStrings => {
+  //When we think user is trying to access help incorrectly, provide them with some helpful information.
+  if (inputStrings.length > 2 && ["help", "--help"].includes(inputStrings[1])) {
+    const help = require("./commands/help").meta;
+    console.log(
+      "Error: please use the following syntax for accessing help information\n",
+      "\n               ",
+      help.help.usage
+    );
+    process.exit();
+  }
+
+  // when `truffle --help <cmd>` is used, convert inputStrings to run as `truffle help <cmd>`
+  if (inputStrings.length > 1 && inputStrings[0] === "--help") {
+    inputStrings[inputStrings.indexOf("--help")] = "help";
+  }
+
+  // when `truffle <cmd> help | --help` is used, convert inputStrings  to run as `truffle help <cmd>`
+  if (["help", "--help"].includes(inputStrings[1])) {
+    inputStrings.pop();
+    inputStrings.unshift("help");
+  }
```

Comment:
since truffle is only looking to support `--help | help` at index[1] and at the end of the position, I didn't want to handle this case here. I `--help | help` is at a location of the commend that truffle doesn't support, we are erroring with the valid ways to use the help comment.

Reply:
Where is that requirement coming from? It's strange to treat a flag and a positional argument with the same because `--help` becomes an alias for 'help`. Not a dealbreaker, I'm curious about the requirement.